lib/improver/tests/utilities/test_TemporalInterpolation.py

```diff
@@ -0,0 +1,234 @@
+
+
+class Test__init__(IrisTest):
+
+    """Test the __init__ method."""
+
+    def test_raises_error_with_no_keyword_args(self):
+        """Test __init__ raises a ValueError if both keywords are unset."""
+        msg = "TemporalInterpolation: One of"
+        with self.assertRaisesRegex(ValueError, msg):
+            TemporalInterpolation()
+
+
+class Test__repr__(IrisTest):
+
+    """Test the repr method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        result = str(TemporalInterpolation(interval_in_minutes=60))
+        msg = '<TemporalInterpolation: interval_in_minutes: 60, times: None>'
+        self.assertEqual(result, msg)
+
+
+class Test_construct_time_list(IrisTest):
+
+    """Test construction of time lists suitable for iris interpolation using
+    this function."""
+
+    def setUp(self):
+        """Set up the test inputs."""
+        self.time_0 = datetime.datetime(2017, 11, 1, 3)
+        self.time_1 = datetime.datetime(2017, 11, 1, 9)
+        self.times = []
+        for i in range(4, 9):
+            self.times.append(datetime.datetime(2017, 11, 1, i))
+        self.expected = [('time', time) for time in self.times]
+
+    def test_list_from_interval_in_minutes(self):
+        """Test generating a list between two times using the
+        interval_in_minutes keyword to define the spacing."""
+
+        result = (
+            TemporalInterpolation(interval_in_minutes=60).construct_time_list(
+                self.time_0, self.time_1))
+        self.assertEqual(self.expected, result)
+
+    def test_non_equally_divisible_interval_in_minutes(self):
+        """Test an exception is raised when trying to generate a list of times
+        that would not divide the time range equally."""
+
+        msg = 'interval_in_minutes provided to time_interpolate does not'
+        with self.assertRaisesRegex(ValueError, msg):
+            TemporalInterpolation(interval_in_minutes=61).construct_time_list(
+                self.time_0, self.time_1)
+
+    def test_list_from_existing_list(self):
+        """Test generating an iris interpolation suitable list from an
+        existing list of datetime objects."""
+
+        result = (
+            TemporalInterpolation(times=self.times).construct_time_list(
+                self.time_0, self.time_1))
+        self.assertEqual(self.expected, result)
+
+    def test_time_list_out_of_bounds(self):
+        """Test an exception is raised when trying to generate a list of times
+        using a pre-existing list that includes times outside the range of the
+        initial and final times."""
+
+        self.times.append(datetime.datetime(2017, 11, 1, 10))
+
+        msg = 'List of times falls outside the range given by'
+        with self.assertRaisesRegex(ValueError, msg):
+            TemporalInterpolation(times=self.times).construct_time_list(
+                self.time_0, self.time_1)
+
+
+class Test_process(IrisTest):
```

Comment:
All these tests take in an `interval_in_minutes` you should probably have one test which uses `times` as input instead.
Also none of these tests have a forecast lead time coordinate associated with the time coordinate, which our real cubes do. It would be good to have a test to show that the forecast lead time coordinate comes out ok too.

Reply:
Forecast period added to all process unit tests.

Reply:
Additional test for using a list.